lib/pbio/drv/usb/stm32_usbd/usbd_pybricks.c

```diff
@@ -0,0 +1,515 @@
+  *                                Pybricks Class  Description
+  *          ===================================================================
+  *
+  *
+  *
+  *
+  *
+  *
+  * @note     In HS mode and when the DMA is used, all variables and data structures
+  *           dealing with the DMA during the transaction process should be 32-bit aligned.
+  *
+  *
+  *  @endverbatim
+  *
+  ******************************************************************************
+  */
+
+/* Includes ------------------------------------------------------------------*/
+#include "usbd_ctlreq.h"
+#include "usbd_pybricks.h"
+
+
+/** @addtogroup STM32_USB_DEVICE_LIBRARY
+  * @{
+  */
+
+
+/** @defgroup USBD_Pybricks
+  * @brief usbd core module
+  * @{
+  */
+
+/** @defgroup USBD_Pybricks_Private_TypesDefinitions
+  * @{
+  */
+/**
+  * @}
+  */
+
+
+/** @defgroup USBD_Pybricks_Private_Defines
+  * @{
+  */
+
+/**
+  * @}
+  */
+
+
+/** @defgroup USBD_Pybricks_Private_Macros
+  * @{
+  */
+
+/**
+  * @}
+  */
+
+
+/** @defgroup USBD_Pybricks_Private_FunctionPrototypes
+  * @{
+  */
+
+static uint8_t USBD_Pybricks_Init(USBD_HandleTypeDef *pdev, uint8_t cfgidx);
+static uint8_t USBD_Pybricks_DeInit(USBD_HandleTypeDef *pdev, uint8_t cfgidx);
+static uint8_t USBD_Pybricks_Setup(USBD_HandleTypeDef *pdev, USBD_SetupReqTypedef *req);
+static uint8_t USBD_Pybricks_DataIn(USBD_HandleTypeDef *pdev, uint8_t epnum);
+static uint8_t USBD_Pybricks_DataOut(USBD_HandleTypeDef *pdev, uint8_t epnum);
+static uint8_t USBD_Pybricks_EP0_RxReady(USBD_HandleTypeDef *pdev);
+static uint8_t *USBD_Pybricks_GetCfgDesc(uint16_t *length);
+
+/**
+  * @}
+  */
+
+/** @defgroup USBD_Pybricks_Private_Variables
+  * @{
+  */
+
+USBD_ClassTypeDef USBD_Pybricks_ClassDriver =
+{
+    .Init = USBD_Pybricks_Init,
+    .DeInit = USBD_Pybricks_DeInit,
+    .Setup = USBD_Pybricks_Setup,
+    .EP0_RxReady = USBD_Pybricks_EP0_RxReady,
+    .DataIn = USBD_Pybricks_DataIn,
+    .DataOut = USBD_Pybricks_DataOut,
+    .GetFSConfigDescriptor = USBD_Pybricks_GetCfgDesc,
+};
+
+/* USB Pybricks device Configuration Descriptor */
+__ALIGN_BEGIN static uint8_t USBD_Pybricks_CfgDesc[USBD_PYBRICKS_CONFIG_DESC_SIZ] __ALIGN_END =
+{
+    /* Configuration Descriptor */
+    0x09,                                     /* bLength: Configuration Descriptor size */
+    USB_DESC_TYPE_CONFIGURATION,              /* bDescriptorType: Configuration */
+    LOBYTE(USBD_PYBRICKS_CONFIG_DESC_SIZ),    /* wTotalLength:no of returned bytes */
+    HIBYTE(USBD_PYBRICKS_CONFIG_DESC_SIZ),
+    0x01,                                     /* bNumInterfaces: 1 interface */
+    0x01,                                     /* bConfigurationValue: Configuration value */
+    0x00,                                     /* iConfiguration: Index of string descriptor describing the configuration */
+    0xC0,                                     /* bmAttributes: self powered */
+    0x32,                                     /* MaxPower 0 mA */
```

Comment:
I couldn't find any good info on this but it seems like we would want to set this to request 500mA for battery charging.